cli/cli_progress.go

```diff
@@ -20,14 +21,19 @@ const (
 )
 
 type progressFlags struct {
-	enableProgress         bool
-	progressUpdateInterval time.Duration
-	out                    textOutput
+	enableProgress              bool
+	progressEstimationType      string
+	adaptiveEstimationThreshold int64
+	progressUpdateInterval      time.Duration
+	out                         textOutput
 }
 
 func (p *progressFlags) setup(svc appServices, app *kingpin.Application) {
 	app.Flag("progress", "Enable progress bar").Hidden().Default("true").BoolVar(&p.enableProgress)
+	app.Flag("progress-estimation-type", "Set type of estimation of the data to be snapshotted").Hidden().Default(snapshotfs.EstimationTypeClassic).
```

Comment:
this and the other parameter should probably be policy-driven, so that each source can define per-source behavior. This way it will also work in the UI/Server mode where passing flags is impossible.

It is ok to have the CLI flags as ultimate overrides on top of policies. 

Having said that, this is self-consistent and we can add policies in separate PR(s).